sim/battle-actions.ts

```diff
@@ -1698,21 +1698,21 @@ export class BattleActions {
 		baseDamage = this.battle.randomizer(baseDamage);
 
 		// STAB
-		if (move.forceSTAB || (type !== '???' && pokemon.hasType(type))) {
+		if (move.forceSTAB || (type !== '???' &&
+			(pokemon.hasType(type) || (pokemon.terastallized && pokemon.getTypes(false, true).includes(type))))) {
```

Comment:
This is an aside, but `getTypes` would be a great place to use an object to pass options — if I'm not already familiar with the function, the arguments in `pokemon.getTypes(false, true)` don't make any sense.